### ISSUE TITLE
rustdoc: Support building docs for examples

### ISSUE THREAD

**Issue by dalcde** [rust-lang/rust#84696]
Examples are often documented but said documentation cannot be rendered as far as I can tell. It is especially useful to use top-level documentation to provide usage notes.

From a presentation point of view, they can be treated as extra crates or listed after modules/macros/functions/etc. In the latter case, it would also serve as an enumeration of examples accompanied with short descriptions.

**Comment by jyn514** [comment]
@dalcde you can do this with `doc = true` under the example.

cc https://github.com/rust-lang/docs.rs/issues/238#issuecomment-724579558

**Comment by dalcde** [comment]
Is there a way to do so without enumerating all examples?

**Comment by jyn514** [comment]
I don't think so, no. This is really a cargo question - @ehuss could you move this to rust-lang/cargo?
